app/javascript/controllers/upload_controller.js

```diff
@@ -141,6 +142,9 @@ export default class extends Controller {
     toggle(this.successTargets, type === "success")
     toggle(this.errorTargets, type === "error")
 
+    // disable form submit while upload in progress
+    this.disableOutlet.uploading(type === "progress")
```

Comment:
i have only been reading about `outlets` but this is my first encounter of it in practice. so in this example, when an upload calls this `show` method, we also want to call the `uploading` method that is in the `disable_controller`?

Reply:
You got it.  "Outlets" are a way for controllers to communicate.  In this case, the UploadController tells the DisableController "hey - an upload started ... increment your count of uploads-in-progress".

Reply:
I messed around with some other ways of accomplishing this... something like...

```html
<div data-disable-target="uploading">
```

And then in the `disable_controller.js`, something like:

```js
uploadingTargetConnected(element) {
  // increment and disable buttons
}

uploadingTargetDisconnected(element) {
  // decrement and enable buttons
}
```

But that was a lot more lines of code.  And I hit a bunch of gotchas... I was having to make the `upload_controller.js` manually add/remove `el.dataset.disable_target = "uploading"`.  The Outlet turned out to be cleaner for 2 JS controllers communicating.